pkgs/development/python-modules/fastexcel/default.nix

```
@@ -0,0 +1,80 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  rustPlatform,
+
+  # nativeBuildInputs
+  cargo,
+  rustc,
+
+  # dependencies
+  pyarrow,
+  typing-extensions,
+
+  # optional-dependencies
+  pandas,
+  polars,
+
+  # tests
+  pytest-mock,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "fastexcel";
+  version = "0.12.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ToucanToco";
+    repo = "fastexcel";
+    tag = "v${version}";
+    hash = "sha256-1BcArjhdbsYZ8VIz1FJYOLKSKQXOjLUXFonIXB+TfiY=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit pname version src;
+    hash = "sha256-JGDNqRF264hNAjQ9bwJnBsQgAcqJjreEbgRZAA58JnY=";
+  };
+
+  nativeBuildInputs = [
+    cargo
+    rustPlatform.cargoSetupHook
+    rustPlatform.maturinBuildHook
+    rustc
+  ];
+
+  dependencies = [
+    pyarrow
+    typing-extensions
+  ];
```

```suggestion
  ] ++ optionals (pythonOlder "3.9") [
    typing-extensions
  ];
```
https://github.com/ToucanToco/fastexcel/blob/fedc19976df12128a60af66995c95a97f08e86dd/pyproject.toml#L29